rasa/nlu/training_data/training_data.py

```diff
@@ -394,44 +435,38 @@ def train_test_split(
         test, train = self.split_nlu_examples(train_frac, random_seed)
 
         # collect all nlg stories
-        test_nlg_stories, train_nlg_stories = self.split_nlg_responses(test, train)
+        test_responses = self._needed_responses_for_examples(test)
+        train_responses = self._needed_responses_for_examples(train)
 
         data_train = TrainingData(
             train,
             entity_synonyms=self.entity_synonyms,
             regex_features=self.regex_features,
             lookup_tables=self.lookup_tables,
-            nlg_stories=train_nlg_stories,
+            responses=train_responses,
         )
-        data_train.fill_response_phrases()
 
         data_test = TrainingData(
             test,
             entity_synonyms=self.entity_synonyms,
             regex_features=self.regex_features,
             lookup_tables=self.lookup_tables,
-            nlg_stories=test_nlg_stories,
+            responses=test_responses,
         )
-        data_test.fill_response_phrases()
 
         return data_train, data_test
 
-    def split_nlg_responses(
-        self, test, train
-    ) -> Tuple[Dict[Text, list], Dict[Text, list]]:
-
-        train_nlg_stories = self.build_nlg_stories_from_examples(train)
-        test_nlg_stories = self.build_nlg_stories_from_examples(test)
-        return test_nlg_stories, train_nlg_stories
-
-    @staticmethod
-    def build_nlg_stories_from_examples(examples) -> Dict[Text, list]:
+    def _needed_responses_for_examples(
+        self, examples: List[Message]
+    ) -> Dict[Text, List[Dict[Text, Any]]]:
+        """Get all responses used in any of the examples."""
 
-        nlg_stories = {}
+        responses = {}
         for ex in examples:
             if ex.get(RESPONSE_KEY_ATTRIBUTE) and ex.get(RESPONSE):
```

Comment:
can no longer happen due to the new validation, can it?
```suggestion
            if ex.get(RESPONSE_KEY_ATTRIBUTE):
```